File: backend/services/auth-service/src/main/java/com/azue/authservice/service/AuthService.java
package com.azue.authservice.service;

import com.azue.authservice.dto.request.LoginRequest;
import com.azue.authservice.dto.request.LogoutRequest;
import com.azue.authservice.dto.request.RefreshTokenRequest;
import com.azue.authservice.dto.request.RegisterRequest;
import com.azue.authservice.dto.response.AuthResponse;
import com.azue.authservice.dto.response.RegisterResponse;

public interface AuthService {

    RegisterResponse register(RegisterRequest request);

    AuthResponse login(LoginRequest request);

    AuthResponse refreshToken(RefreshTokenRequest request);

    void logout(LogoutRequest request);

}

File: frontend/azen/src/app/core/auth/auth.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { map } from 'rxjs';
import { ConfigService } from '../config/config.service';

import {
  AuthResponse,
  LoginRequest,
  RegisterRequest
} from './auth.types';

@Injectable({
  providedIn: 'root'
})
export class AuthService {
    private readonly config = inject(ConfigService);

  private readonly http = inject(HttpClient);

  private readonly apiUrl =
  `${this.config.apiUrl}/api/v1/auth`;

  login(credentials: LoginRequest): Observable<AuthResponse> {
    return this.http.post<AuthResponse>(
      `${this.apiUrl}/login`,
      credentials
    );
  }

  register(data: RegisterRequest): Observable<void> {
    return this.http.post(
        `${this.apiUrl}/register`,
        data,
        {
        responseType: 'text'
        }
    ).pipe(
        map(() => void 0)
    );
    }

  refreshToken(
    refreshToken: string
  ): Observable<AuthResponse> {
    return this.http.post<AuthResponse>(
      `${this.apiUrl}/refresh-token`,
      { refreshToken }
    );
  }
}